homeassistant/components/emulated_hue.py

```diff
@@ -0,0 +1,623 @@
+        # This is primarily useful for things like scenes or scripts, which
+        # don't really have a concept of being off
+        self.off_maps_to_on_domains = conf.get(CONF_OFF_MAPS_TO_ON_DOMAINS)
+        if not isinstance(self.off_maps_to_on_domains, list):
+            self.off_maps_to_on_domains = DEFAULT_OFF_MAPS_TO_ON_DOMAINS
+
+        # Get whether or not entities should be exposed by default, or if only
+        # explicitly marked ones will be exposed
+        self.expose_by_default = conf.get(CONF_EXPOSE_BY_DEFAULT)
+        if not isinstance(self.expose_by_default, bool):
+            self.expose_by_default = DEFAULT_EXPOSE_BY_DEFAULT
+
+        # Get domains that are exposed by default when expose_by_default is
+        # True
+        self.exposed_domains = conf.get(CONF_EXPOSED_DOMAINS)
+        if not isinstance(self.exposed_domains, list):
+            self.exposed_domains = DEFAULT_EXPOSED_DOMAINS
+
+
+class DescriptionXmlView(HomeAssistantView):
+    """Handles requests for the description.xml file."""
+
+    url = '/description.xml'
+    name = 'description:xml'
+    requires_auth = False
+
+    def __init__(self, hass, config):
+        """Initialize the instance of the view."""
+        super().__init__(hass)
+        self.config = config
+
+    def get(self, request):
+        """Handle a GET request."""
+        xml_template = """<?xml version="1.0" encoding="UTF-8" ?>
+<root xmlns="urn:schemas-upnp-org:device-1-0">
+<specVersion>
+<major>1</major>
+<minor>0</minor>
+</specVersion>
+<URLBase>http://{0}:{1}/</URLBase>
+<device>
+<deviceType>urn:schemas-upnp-org:device:Basic:1</deviceType>
+<friendlyName>Home Assistant Emulated Hue Bridge ({0})</friendlyName>
+<manufacturer>Royal Philips Electronics</manufacturer>
+<manufacturerURL>http://www.philips.com</manufacturerURL>
+<modelDescription>Philips hue Personal Wireless Lighting</modelDescription>
+<modelName>Philips hue bridge 2015</modelName>
+<modelNumber>BSB002</modelNumber>
+<modelURL>http://www.meethue.com</modelURL>
+<serialNumber>1234</serialNumber>
+<UDN>uuid:2f402f80-da50-11e1-9b23-001788255acc</UDN>
+</device>
+</root>
+"""
+
+        resp_text = xml_template.format(
+            self.config.host_ip_addr, self.config.listen_port)
+
+        return self.Response(resp_text, mimetype='text/xml')
+
+
+class HueUsernameView(HomeAssistantView):
+    """Handle requests to create a username for the emulated hue bridge."""
+
+    url = '/api'
+    name = 'hue:api'
+    requires_auth = False
+
+    def __init__(self, hass):
+        """Initialize the instance of the view."""
+        super().__init__(hass)
+
+    def post(self, request):
+        """Handle a POST request."""
+        data = request.json
+
+        if 'devicetype' not in data:
+            return self.Response("devicetype not specified", status=400)
+
+        json_response = [{'success': {'username': '12345678901234567890'}}]

```

Comment:
For now, we can leave it as is in the interest of getting this merged before the cutoff. However, in the future, it should randomly generate a string and store that string in a flat file so that only actually authorized usernames can get through.